mmv1/third_party/terraform/services/container/node_config.go.tmpl

```diff
@@ -2350,6 +2446,34 @@ func nodePoolNodeConfigUpdate(d *schema.ResourceData, config *transport_tpg.Conf
 				req.StoragePools = storagePools
 			}
 
+			if v, ok := d.GetOk("node_config.0.boot_disk"); ok {
```

Comment:
I'm getting into a weird permadiff loop here. If I have a config that specifies both of these fields with different values the apply "succeeds" but the request is rejected with a 400 error complaining that both are set with different values.

It seems like we aren't correctly checking the result of the update call in this case.

Would it make sense to prevent users from specifying both versions of the field, like have a conflict between `disk_size_gb` and `boot_disk.size_gb` to prevent this?

Reply:
That is intentional; the API will reject it with an error message if the two values are set differently.

We could make a change in the terraform code to handle this, but the goal was to allow as much of the logic as possible to live in the backend, since the client code is difficult to change in backwards compatible ways. If you think it's a blocker (i.e. the error isn't clear enough to the user that those need to be set similarly), we can discuss options.